src/core/reducers/abuse.js

```diff
@@ -0,0 +1,81 @@
+/* @flow */
+import type { ErrorHandlerType } from 'core/errorHandler';
+
+
+export const LOAD_ADDON_ABUSE_REPORT = 'LOAD_ADDON_ABUSE_REPORT';
+export const SEND_ADDON_ABUSE_REPORT = 'SEND_ADDON_ABUSE_REPORT';
+
+type LoadAddonAbuseReportType = {
+  addon: {|
+    guid: string,
+    id: number,
+    slug: string,
+  |},
+  message: string,
+  reporter: Object | null,
+};
+
+export function loadAddonAbuseReport(
+  { addon, message, reporter }: LoadAddonAbuseReportType
+) {
+  if (!addon) {
+    throw new Error('addon is required');
+  }
+  if (!message) {
+    throw new Error('message is required');
+  }
+  if (reporter === undefined) {
+    throw new Error('reporter cannot be undefined');
+  }
+
+  return {
+    type: LOAD_ADDON_ABUSE_REPORT,
+    payload: { addon, message, reporter },
+  };
+}
+
+type SendAddonAbuseReportAction = {|
+  addonSlug: string,
+  errorHandler: ErrorHandlerType,
+  message: string,
+|};
+
+export function sendAddonAbuseReport(
+  { addonSlug, errorHandler, message }: SendAddonAbuseReportAction
+) {
+  if (!addonSlug) {
+    throw new Error('addonSlug is required');
+  }
+  if (!errorHandler) {
+    throw new Error('errorHandler is required');
+  }
+  if (!message) {
+    throw new Error('message is required');
+  }
+
+  return {
+    type: SEND_ADDON_ABUSE_REPORT,
+    payload: { addonSlug, errorHandlerId: errorHandler.id, message },
+  };
+}
+
+export const initialState = {};
+
+type ReducerState = {};
+
+export default function abuseReducer(
+  state: ReducerState = initialState,
+  action: Object
+) {
+  switch (action.type) {
+    case LOAD_ADDON_ABUSE_REPORT: {
+      const { addon, message, reporter } = action.payload;
+      return {
+        ...state,
+        [addon.slug]: { message, reporter },
```

Comment:
Hah, so I will try to give the cliffnotes version here but I can explain more on video or IRC if you like...

The current abuse reports aren't stored anywhere. Or rather, they aren't stored in the API or even in a DB if I recall correctly. They're emailed to admins and people take a look. So we're storing them essentially to prevent the same user submitting multiple abuse reports in a single session. But that's it for now. They aren't stored anywhere so I'm just storing them here because it seems the right thing to do *and* I can refuse multiple reports per-session. But yeah, that's about it. The abuse reporting features are pretty under-developed 😢

Reply:
> I'm just storing them here because ... I can refuse multiple reports per-session

In that case, maybe a lookup structure keyed by add-on ID and reporter?

```js
return {
  ...state,
  bySlug: {
    ...state.bySlug,
    [addon.slug]: {
      ...state.bySlug[addon.slug],
      [reporter]: message,
    },
  },
};
```

We should definitely only build state for what we need :)